lib/spack/spack/compilers/adaptor.py

```diff
@@ -79,6 +79,22 @@ def implicit_rpaths(self) -> List[str]:
             result.extend(CompilerPropertyDetector(compiler).implicit_rpaths())
         return result
 
+    def _first_defined(self, property):
+        for language in [Languages.C, Languages.CXX, Languages.FORTRAN]:
+            if language in self.compilers:
+                compiler_pkg = self.compilers[language].package
+                opt_flags = getattr(compiler_pkg, property, None)
+                if opt_flags:
+                    return opt_flags
+
+    @property
+    def opt_flags(self) -> List[str]:
+        return self._first_defined("opt_flags") or []
+
+    @property
+    def debug_flags(self) -> List[str]:
+        return self._first_defined("debug_flags") or []
```

Comment:
I think here and above you can use:
```suggestion
        return next(iter(self.compilers.values())).package.opt_flags
```
The constructor ensures that there is at least one value, while:

https://github.com/spack/spack/blob/1904d99fd0fc592978cbc8e84321d723a8983a31/lib/spack/spack/compilers/adaptor.py#L202-L213

ensure the dict is constructed with C, followed by C+, followed by Fortran.